src/components/UriViewer.js

```diff
@@ -52,7 +68,38 @@ const getMaxDownloadCostNA = bytes => {
   return downloadPrice < 0.01 ? '< $0.01' : Utils.formatUSD(downloadPrice)
 }
 
-const UriViewer = ajaxCaller(class UriViewer extends Component {
+const UriViewer = props => {
+  const signal = useCancellation()
+  const [valid, setValid] = useState(undefined)
+
+  const { googleProject, uri, onDismiss } = props
+  const isGsUri = isGs(uri)
+  const [bucket, name] = isGsUri ? parseUri(uri) : []
+
+  const pingObject = withErrorReporting('Error fetching file metadata', async () => {
+    if (isGsUri) {
+      try {
+        await Ajax(signal).Buckets.getObject(bucket, name, googleProject)
+        setValid(true)
+      } catch (e) {
+        const isRequesterPays = e.status === 400 && (await e.json()).error.message.includes('requester pays')
+        setValid(!isRequesterPays)
+      }
+    }
+  })
+
+  Utils.useOnMount(() => {
+    pingObject()
+  })
+
+  return Utils.switchCase(valid,
+    [undefined, () => h(Modal, viewerProps(onDismiss), loadingMetadata(uri))],
+    [true, () => h(UriViewerModal, props)],
+    [false, () => h(Modal, viewerProps(onDismiss), ['Error: The file is in a requester pays bucket but there is no user project provided.'])]
```

Comment:
Please provide feedback/suggestions for improving the language for the error modal!

Reply:
```suggestion
    [false, () => h(Modal, viewerProps(onDismiss), ['Error: To view this file you must provide a billing project, because this file is in a requester pays Google Bucket. For more information [add link].'])]
```

Reply:
Do we have documentation that we can link out to about requester pays buckets?

Reply:
Not AFAIK. Whoever is championing this issue (you? 😄 ) should engage with the Content strategists to put this on their radar. It shouldn't hold up merging, but we should add it as a fast-follow.

Reply:
I'll make a ticket and poke comms to put it on their radar, but I won't have it hold up merging!

Reply:
https://broadworkbench.atlassian.net/browse/SATURN-173